script/util/copy-verifier.sh

```diff
@@ -4,99 +4,48 @@ GIT_REPO_PATH=../mapreduce-plonky2
 
-cd $GIT_REPO_PATH && \
-    git fetch origin $BRANCH && \
-    git checkout $BRANCH && \
-    git pull origin $BRANCH && \
-    cd -
-
 mkdir -p $VERIFIER_FOLDER
 
 wget -O $VERIFIER_FILE $VERIFIER_SOL_URL
-cp "${GIT_REPO_PATH}/${CODE_DIR_PATH}/Groth16VerifierExtensions.sol" $VERIFIER_EXTENSIONS_FILE
-
-# Use awk for the following transformations
-awk '{
-  # Use as library instead of contract
-  gsub(/contract Verifier/, "library Groth16Verifier");
-
-  # Use internal instead of public functions
-  gsub(/public view/, "internal view");
-
-  # Read proof argument from memory instead of calldata
-  gsub(/calldata proof/, "memory proof");
-  gsub(/calldatacopy\(f, proof, 0x100\)/, "mstore(f, mload(add(proof, 0x00)))\nmstore(add(f, 0x20), mload(add(proof, 0x20)))\nmstore(add(f, 0x40), mload(add(proof, 0x40)))\nmstore(add(f, 0x60), mload(add(proof, 0x60)))\nmstore(add(f, 0x80), mload(add(proof, 0x80)))\nmstore(add(f, 0xa0), mload(add(proof, 0xa0)))\nmstore(add(f, 0xc0), mload(add(proof, 0xc0)))\nmstore(add(f, 0xe0), mload(add(proof, 0xe0)))");
-
-  # Read input argument from memory instead of calldata
-  gsub(/calldata input/, "memory input");
-  gsub(/calldataload\(input\)/, "mload(input)");
-  gsub(/calldataload\(add\(input, 32\)\)/, "mload(add(input, 32))");
-  gsub(/calldataload\(add\(input, 64\)\)/, "mload(add(input, 64))");
-
-  print;
-}' $VERIFIER_FILE > $VERIFIER_FILE.tmp && mv $VERIFIER_FILE.tmp $VERIFIER_FILE
+wget -O $VERIFIER_EXTENSION_FILE $VERIFIER_EXTENSION_URL
 
+# TODO - remove all manual transformations... so close!
 awk '{
   # Import verifier library with renamed filename
-  gsub(/import {Verifier} from ".\/verifier.sol";/, "import {Groth16Verifier} from \".\/Groth16Verifier.sol\";\n   import {isCDK} from \"..\/utils\/Constants.sol\";");
-
-  # Use extensions as library instead of contract
-  gsub(/contract Query is Verifier {/, "library Groth16VerifierExtensions {");
-  gsub(/CIRCUIT_DIGEST/, "Groth16Verifier.CIRCUIT_DIGEST");
-  gsub(/this.verifyProof/, "Groth16Verifier.verifyProof");
-
-  # Use internal instead of public functions
-  gsub(/public view/, "internal view");
-
-  # Change "view" to "pure" in "function verifyQuery"
-  if (match($0, /function verifyQuery.*/)) {
-    sub(/pure/, "view"); 
-  }
+  gsub(/import {Verifier} from ".\/Verifier.sol";/, "import {Verifier} from \".\/Verifier.sol\";\n   import {isCDK} from \"..\/utils\/Constants.sol\";");
 
   # Patch `verifyQuery` function to skip blockhash verification for polygon CDK chains
-  gsub(/blockHash == query.blockHash/, "isCDK() || blockHash == query.blockHash");
+  gsub(/blockHash == expectedBlockHash/, "isCDK\(\) || blockHash == expectedBlockHash");
 
   print;
-}' $VERIFIER_EXTENSIONS_FILE > $VERIFIER_EXTENSIONS_FILE.tmp && mv $VERIFIER_EXTENSIONS_FILE.tmp $VERIFIER_EXTENSIONS_FILE
-
+}' $VERIFIER_EXTENSION_FILE > $VERIFIER_EXTENSION_FILE.tmp && mv $VERIFIER_EXTENSION_FILE.tmp $VERIFIER_EXTENSION_FILE
```

Comment:
The only manual modification left is the injection of `isCDK()` - I plan to remove this in a future PR and we'll be rid of the manual code generation.